Userland/passwd.cpp

```diff
@@ -105,41 +71,36 @@ int main(int argc, char** argv)
 
     uid_t current_user = getuid();
 
-    struct passwd pwd;
+    auto account = Core::Account::get(current_user).value();
     if (username) {
-        pwd = *getpwnam(username);
-        if (current_user != 0 && current_user != pwd.pw_uid) {
+        auto result = Core::Account::get(username);
```

Comment:
To be less confusing, I'd add comments (and more descriptive variable names) that the above is the account for the current user, and this is for the user we're modifying.